src/Setup/WixBA/RootView.xaml

```diff
@@ -1,125 +1,487 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information. -->
-
+<!--  Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information.  -->
 
 <Window x:Class="WixToolset.UX.RootView"
-    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
-    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-    xmlns:ux="clr-namespace:WixToolset.UX"
-    WindowStartupLocation="CenterScreen"
-    WindowStyle="None"
-    AllowsTransparency="True"
-    Background="{x:Null}"
-    Width="400"
-    Height="400">
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:i="http://schemas.microsoft.com/expression/2010/interactivity"
+        xmlns:ux="clr-namespace:WixToolset.UX"
+        MinWidth="550"
+        MinHeight="400"
+        MaxWidth="1200"
```

Comment:
I think I pointed this out in the PR, but the issue is that we added a
WebBrowser control that is displayed when the UpdateViewModel detects an
available update.  The WPF WebBrowser control is notorious for having
layout problems, and from what I read there is no real solution to these
issues because it is a thin wrapper around a win32 com object.  So when
there was not min and max defined and the user was allowed to resize beyond
these ranges the WebBrowser Control on reduction gets pushed over the top
of other layout and on expansion above these limits it is less problematic
but also seemed to be rather ugly.

Yes it was proposed in the WIP review that we support user resizing and a
Max button.  I spent a lot of time on the above issue, and even tried for a
while to go down the path of eliminating the WebBrowser control and
implementing a solution using a FlowDocument, which was the recommendation
that I found when researching the problem.  That however requires importing
a HTML to FlowDocument conversion.  I was not clear on the license impacts
of the code that I found in this regard so I decided not to pursue that
path any further at this time.  And in the WIP review it was proposed that
maybe someone else would want to work on this aspect of the changes.

This is also the reason for the need for Areo No Snap.  The Max window
button is available, which maximizes to the MaxWidth which we defined so
that the above layout is not messed up.

The way that I tested this behavior, since I did not have access to staging
an available update, was to hack src/Setup/Bundle/Bundle.wxs to have a
lower version, such as 3.9.0.0, and then hack
src/common/WixDistribution.wxl to point at the 3.10 feed.  This allows the
bundle to detect a staged update and display the details.  I also used test
code in UpdateViewModel/DetectUpdates to insert more these into
this.UpdateChanges to test the layout with a large HTML string.

So I think there are competing goals here and I don't know how at this time
to please all perspectives.  Feel free to let me know your view or how
these issues should be prioritized.

On Sun, Jul 31, 2016 at 11:51 PM, Sean Hall <notifications@github.com>
wrote:

> In src/Setup/WixBA/RootView.xaml
> <https://github.com/wixtoolset/wix4/pull/208#discussion_r72923919>:
>
> > -    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
> > -    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
> > -    xmlns:ux="clr-namespace:WixToolset.UX"
> > -    WindowStartupLocation="CenterScreen"
> > -    WindowStyle="None"
> > -    AllowsTransparency="True"
> > -    Background="{x:Null}"
> > -    Width="400"
> > -    Height="400">
> > +        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
> > +        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
> > +        xmlns:i="http://schemas.microsoft.com/expression/2010/interactivity"
> > +        xmlns:ux="clr-namespace:WixToolset.UX"
> > +        MinWidth="550"
> > +        MinHeight="400"
> > +        MaxWidth="1200"
>
> If it's maximizable, there shouldn't be a MaxWidth or MaxHeight
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wixtoolset/wix4/pull/208/files/28721ab8e4985a5641b3efd7975a67cecd8f8d4a#r72923919>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyCLbEPh3-BgixU82UqQpHMNoY3sRFzks5qbXtRgaJpZM4JVm8P>
> .
>